### PR TITLE
Prevent resetting latest flag of real-time when starting historical analysis

### PR DESCRIPTION
**Pull request by kaituo**
### Description
This PR addresses a bug where starting a historical analysis after a real-time analysis on the same detector caused the real-time task’s latest flag to be incorrectly reset to false by the historical run.

The fix ensures that only the latest flags of the same analysis type are reset:
* Real-time analysis will only reset the latest flag of previous real-time analyses.
* Historical analysis will only reset the latest flag of previous historical analyses.

Additional Changes:
* Set a minimum recencyEmphasis value of 2 to align with RCF requirements.
* Updated imputation logic to execute only when the entity maps to the local node ID in the hash ring, reducing errors when models of the same entity exist on multiple nodes.

Testing:
- Added an integration test to reproduce the bug and verified the fix.

### Check List
- [X] New functionality includes testing.
- [X] New functionality has been documented.
- [X] API changes companion pull request [created](https://github.com/opensearch-project/opensearch-api-specification/blob/main/DEVELOPER_GUIDE.md).
- [X] Commits are signed per the DCO using `--signoff`.
- [X] Public documentation issue/PR [created](https://github.com/opensearch-project/documentation-website/issues/new/choose).

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/anomaly-detection/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

File: src/main/java/org/opensearch/ad/transport/ADHCImputeTransportAction.java [hunk 1]
```diff
@@ -156,4 +159,46 @@ protected ADHCImputeNodeResponse nodeOperation(ADHCImputeNodeRequest nodeRequest
...
+    /**
+     * Determines whether the model state should be processed based on various conditions.
+     *
+     * Conditions checked:
+     * - The model's last seen execution end time is not the minimum Instant value.
+     * - The current execution end time is greater than or equal to the model's last seen execution end time,
+     *   indicating that the model state was updated in previous intervals.
+     * - The entity associated with the model state is present.
+     * - The owning node for real-time processing of the entity, with the same local version, is present in the hash ring.
+     * - The owning node for real-time processing matches the current local node.
```
Comment (amitgalitz) [inline src/main/java/org/opensearch/ad/transport/ADHCImputeTransportAction.java:171]: before we could have occasions where two nodes have the same model for the same entity, so two nodes at the same time will find if there is an anomaly on the interval?

Comment (kaituo) [reply src/main/java/org/opensearch/ad/transport/ADHCImputeTransportAction.java:171]: yes.

Comment (amitgalitz) [reply src/main/java/org/opensearch/ad/transport/ADHCImputeTransportAction.java:171]: would this mean we would write result twice or we handled it later to only be written once?

**Review by amitgalitz — APPROVED** [review]

File: src/main/java/org/opensearch/ad/transport/ADHCImputeTransportAction.java (hunk 1 above)
Comment (kaituo) [reply src/main/java/org/opensearch/ad/transport/ADHCImputeTransportAction.java:171]: We would only write it once (not guaranteed) as here we only process imputation on a node the entity is mapped to.

**Pull request merged**